pallets/ddc-verification/src/lib.rs

```diff
@@ -1651,42 +1660,50 @@ pub mod pallet {
 			for bucket_node_aggregate_activity in bucket_node_aggregates_not_in_consensus {
 				let merkle_node_ids = Self::find_random_merkle_node_ids(
 					number_of_identifiers.into(),
-					bucket_node_aggregate_activity.clone(),
+					bucket_node_aggregate_activity.number_of_gets +
+						bucket_node_aggregate_activity.number_of_puts,
+					bucket_node_aggregate_activity.clone().node_id,
 				);
 				let bucket_id = bucket_node_aggregate_activity.clone().bucket_id;
 
 				log::info!("🚀 Merkle Node Identifiers for bucket_node_aggregate_activity: node id: {:?} bucket_id:{:?}  identifiers: {:?}",
 						bucket_node_aggregate_activity.clone().node_id, bucket_id.clone(), merkle_node_ids);
+				let pub_key = NodePubKey::StoragePubKey(Self::get_node_pub_key(
+					bucket_node_aggregate_activity.clone().node_id,
+				));
```

Comment:
`node_id` you are referring to here is the identifier of the node where there was partial activity of a bucket.
It has nothing in common with the `/challenge` endpoint.

The `/challenge` endpoint is supposed to challenge the Aggreagtor that returned you the bucket sub-aggregate which is not in consensus.

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/5f9534cb-39ab-4238-b9ae-ce751179cb61">